class-based/src/say-function.ts

```diff
@@ -0,0 +1,22 @@
+import { Construct } from "constructs";
+import { Function } from "functionless";
+
+interface InputEvent {
+  message: string;
+}
+
+export class SayFunction extends Construct {
+  fn: Function<InputEvent, void>;
+
+  constructor(scope: Construct, id: string) {
+    super(scope, id);
+    this.fn = new Function(
+      scope,
+      "SayFunction",
```

Comment:
This is a problem area I've run into in the past. This is going to result in the id path being `SayFunction/SayFunction` which usually will duplicate the name in the created resource. This is the CDK-recommended way of making a construct, even if it is just one resource, but the double naming leads to a clunky experience.

Reply:
This is up to the user though. If the caller determine the meaning, then the caller should call the top level "sayFunction". If the construct determines the meaning, the construct should call it "sayFunction".

In the form case, the construct can just call it "f" or something to leave it up to the caller. Or "Resource" like CDK does.